engine/src/EntityManager.h

```diff
@@ -0,0 +1,47 @@
+//
+// Created by tdancois on 8/4/16.
+//
+
+#ifndef ARES_ENTITYMANAGER_H
+#define ARES_ENTITYMANAGER_H
+
+
+#include <memory>
+#include <map>
+#include "Entity.h"
+#include "AnimatedSprite.h"
+#include "AnimatedSpritesUpdater.h"
+#include <SFML/Graphics/RenderTarget.hpp>
+#include "Player.h"
+
+class EntityManager {
+//    singleton
+    static EntityManager *instance;
+
+    EntityManager();
+
+    std::map<unsigned int, std::shared_ptr<Entity>> entities;
+    unsigned int player_id;
+    bool isPlayerSet;
+public:
+    static EntityManager *getInstance();
+
+    void addNewPlayer(const unsigned int &id, const std::string &character_texture);
```

Comment:
The only exception to this rule is for smart pointers because making a copies are the heart of smart pointers mechanism, with reference counter (which in fact count only copies...)